decbinhex4/addons/movement_demo/src/code/polygon_class.nim

```diff
@@ -0,0 +1,21 @@
+import gdext, gdextgen/classes/gdSceneTree
+import gdextgen/classes / [gdInput, gdPolygon2D]
+
+type PolygonClass* = ref object of Polygon2D
+  Input:Input
+  InputMap:InputMap
+
+method ready*(self:PolygonClass) {.gdsync.} =
+  if isRunningInEditor: return
+  print("this will print even if the class is hidden, and esc will still exit")
```

Comment:
Perhaps the names of the class defined internally for this determination conflict. I will fix this soon.

Reply:
I also tried it with a rename to `MovingPolygon` and it had the same behavior.